WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/FollowerTypesUseCase.kt

```diff
@@ -0,0 +1,146 @@
+        }
+        socialTotals?.let {
+            if (it > 0) {
+                map[SOCIAL] = it
+            }
+        }
+        return map
+    }
+
+    override suspend fun fetchRemoteData(forced: Boolean) = fetchData(forced, PagedMode(0, false))
+
+    private suspend fun fetchData(forced: Boolean, fetchMode: PagedMode): State<Map<FollowerType, Int>> {
+        val deferredWpComResponse = async(bgDispatcher) {
+            followersStore.fetchWpComFollowers(statsSiteProvider.siteModel, fetchMode, forced)
+        }
+        val deferredEmailResponse = async(bgDispatcher) {
+            followersStore.fetchEmailFollowers(statsSiteProvider.siteModel, fetchMode, forced)
+        }
+        val deferredPublicizeResponse = async(bgDispatcher) {
+            publicizeStore.fetchPublicizeData(statsSiteProvider.siteModel, LimitMode.All, forced)
+        }
+
+        val wpComResponse = deferredWpComResponse.await()
+        val emailResponse = deferredEmailResponse.await()
+        val publicizeResponse = deferredPublicizeResponse.await()
+
+        val wpComModel = wpComResponse.model
+        val emailModel = emailResponse.model
+        val socialTotals = publicizeResponse.model?.services?.sumOf { it.followers }
+
+        val error = wpComResponse.error ?: emailResponse.error ?: publicizeResponse.error
+        val data = buildDataModel(wpComModel?.totalCount, emailModel?.totalCount, socialTotals)
+        return when {
+            error != null -> State.Error(error.message ?: error.type.name)
+            data.isNotEmpty() -> State.Data(data)
+            else -> State.Empty()
+        }
+    }
+
+    private fun getTitle(type: FollowerType) = when (type) {
+        WP_COM -> R.string.stats_followers_wordpress_com
+        EMAIL -> R.string.email
+        SOCIAL -> R.string.stats_insights_social
+    }
+
+    override fun buildUiModel(domainModel: Map<FollowerType, Int>): List<BlockListItem> {
+        val items = mutableListOf<BlockListItem>()
+        if (domainModel.isNotEmpty()) {
+            val total = domainModel.entries.sumOf { it.value }
+
+            domainModel.entries.forEach {
+                val title = getTitle(it.key)
+                val value = buildValuePercent(it.value, total)
+                items.add(
+                        ListItem(
+                                text = resourceProvider.getString(title),
+                                value = value,
+                                showDivider = domainModel.entries.indexOf(it) < domainModel.size - 1,
+                                contentDescription = contentDescriptionHelper.buildContentDescription(title, it.value)
+                        )
+                )
+            }
+        } else {
+            items.add(Empty())
+        }
+        return items
+    }
+
+    private fun buildValuePercent(value: Int, total: Int): String {
+        val percentage = if (total == 0) {
+            0
+        } else {
+            value * 100 / total
```

Comment:
When total is above 100 then the % will be a fraction.  For e.g. 2 * 100 / 629 = 0.31
Need to handle case for below 1% (  ~1%) or show it as fractional  like 0.3% in this case using formatted string.

Reply:
Okay, I implement your advice. I'm showing the floating percentage value if it's below 1.0. f3df9425647c2ec9e4582b5b58824749cc3217d3